### PR TITLE
feat(knock requests) : branch logic for handling knock requests

### PR DESCRIPTION
**Pull request by ganfra**
## Content
Branch all the logic when displaying banner and knock requests list : 
- allow to dismiss the banner for the currently seen requests
- allow to accept/decline requests and ban users
- everything is live thanks to the sdk

## Motivation and context
https://github.com/element-hq/element-x-android/issues/3692

## Tests

- Create a knockable room
- Open the room
- With another account (on the same HS) navigate to this room (through permalink or room directory)
- Use the knock button
- See the request from the admin account, and play with the actions

## Tested devices

- [X] Physical
- [X] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

**Review by bmarty — APPROVED** [review]
Nice work, thanks!

File: features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/data/KnockRequestsService.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            .map { knockRequest ->
+                async {
+                    acceptKnockRequest(knockRequest, optimistic = optimistic)
+                }
+            }
+            .awaitAll()
+        if (results.all { it.isSuccess }) {
+            Result.success(Unit)
+        } else {
+            Result.failure(IllegalStateException("Failed to accept all knock requests"))
```
[the remark this answers is not in the document]
Comment (ganfra) [reply features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/data/KnockRequestsService.kt:108]: We are using custom dialogs, so not using the content of the error, but sure I can create some exception

File: features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/list/KnockRequestsListView.kt [hunk 2]
```diff
@@ -336,7 +420,7 @@ private fun KnockRequestItem(
                 TextButton(
                     text = stringResource(R.string.screen_knock_requests_list_decline_and_ban_action_title),
                     onClick = {
-                        onAcceptClick(knockRequest)
+                        onBanClick(knockRequest)
```
[the remark this answers is not in the document]
Comment (ganfra) [reply features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/list/KnockRequestsListView.kt:423]: It's not the current state, right?

**Review by bmarty — APPROVED** [review]
Thanks for the update!

**Pull request merged**